tests/integration/test_random_creation.py

```diff
@@ -33,6 +39,447 @@ def test_randn():
+    )
+
+
+LOW_HIGH = [
+    (5, 10000),
+    (-10000, 5),
+]
+SMALL_RNG_SIZES = [5, 1024, (1, 2)]
+LARGE_RNG_SIZES = [10000, (20, 50, 4)]
+ALL_RNG_SIZES = SMALL_RNG_SIZES + LARGE_RNG_SIZES
+INT_DTYPES = [np.int64, np.int32, np.int16]
+UINT_DTYPES = [np.uint64, np.uint16, np.uint0]
+
+
+@pytest.mark.parametrize("size", ALL_RNG_SIZES, ids=str)
+@pytest.mark.parametrize("low, high", LOW_HIGH, ids=str)
+@pytest.mark.parametrize("dtype", INT_DTYPES, ids=str)
+def test_randint_basic_stats(low, high, size, dtype):
+    arr_np, arr_num = gen_random_from_both(
+        "randint", low=low, high=high, size=size, dtype=dtype
+    )
+    assert arr_np.dtype == arr_np.dtype
+    assert arr_np.shape == arr_num.shape
+    assert np.min(arr_num) >= low
+    assert np.max(arr_num) <= high
+
+
+@pytest.mark.parametrize("low", [1024, 1025, 12345], ids=str)
+@pytest.mark.parametrize("size", LARGE_RNG_SIZES, ids=str)
+@pytest.mark.parametrize("dtype", INT_DTYPES, ids=str)
+def test_randint_low_limit_only(low, size, dtype):
+    arr_np, arr_num = gen_random_from_both(
+        "randint", low, size=size, dtype=dtype
+    )
+    assert arr_np.dtype == arr_np.dtype
+    assert arr_np.shape == arr_num.shape
+    assert_distribution(
+        arr_num, np.mean(arr_np), np.std(arr_np), mean_tol=0.05
+    )
+
+
+@pytest.mark.xfail(
+    EAGER_TEST,
+    reason="cuNumeric randint high limit is inclusive in Eager mode",
+)
+def test_randint_high_limit():
+    limit = 10
+    arr_np, arr_num = gen_random_from_both("randint", 10, size=100)
+    assert np.max(arr_np) < limit
+    assert np.max(arr_num) < limit
+
+
+@pytest.mark.xfail(not EAGER_TEST, reason="cuNumeric hits struct error.")
+@pytest.mark.parametrize(
+    "low, high", [(3000.45, 15000), (123, 456.7), (12.3, 45.6)], ids=str
+)
+def test_randint_float_range(low, high):
+    # When size is greater than 1024, legate core throws error
+    # struct.error: required argument is not an integer
+    arr_np, arr_num = gen_random_from_both(
+        "randint", low=low, high=high, size=1025
+    )
+    assert_distribution(
+        arr_num, np.mean(arr_np), np.std(arr_np), mean_tol=0.05
+    )
+
+
+@pytest.mark.xfail(
+    not EAGER_TEST, reason="cuNumeric raises NotImplementedError"
+)
+@pytest.mark.parametrize("size", ALL_RNG_SIZES, ids=str)
+@pytest.mark.parametrize("low, high", [(1000, 65535), (0, 1024)], ids=str)
+@pytest.mark.parametrize("dtype", UINT_DTYPES, ids=str)
+def test_randint_uint(low, high, dtype, size):
+    # NotImplementedError: type for random.integers has to be int64 or int32
+    # or int16
+    arr_np, arr_num = gen_random_from_both(
+        "randint", low=low, high=high, size=size, dtype=dtype
+    )
+    assert arr_np.dtype == arr_np.dtype
+    assert arr_np.shape == arr_num.shape
+    assert np.min(arr_num) >= low
+    assert np.max(arr_num) <= high
+
+
+@pytest.mark.xfail(
+    reason="NumPy returns scalar, cuNumeric returns 1-dim array"
```

Comment:
@manopapad this (returning singleton arrays vs scalars) is always going to be the case, correct? If so I think we can just codify the expected behavior and not that it is an expected difference from numpy.

Reply:
In this particular case we have no problem returning a scalar, we were just setting the wrong default shape, `(1,)` instead of `()`. I fixed this with https://github.com/nv-legate/cunumeric/pull/966.

Reply:
@yimoj I think this test can be updated as well, now that #966 is merged?

Reply:
It's still hitting XFAIL condition though...
https://pipelines.actions.githubusercontent.com/serviceHosts/7bae2b68-cf8e-4a4e-83c2-ad2e6b2bf3f8/_apis/pipelines/1/runs/7076/signedlogcontent/20?urlExpires=2023-06-21T02%3A48%3A04.2784311Z&urlSigningMethod=HMACV1&urlSignature=YScGAQLPpFtY0w262HdaK8yRESaY%2BfwZQvDBjVU4jU0%3D

> 2023-06-14T10:27:30.8639870Z    tests/integration/test_random_creation.py::test_randint_size_none XFAIL  [ 64%]

```
>>> num.random.randint(1234, size=None)
array(623)
```

Reply:
cc @manopapad

Reply:
It works on my local machine. The following code:

```
x = cn.random.randint(1234, size=None)
print(x)
print(type(x))
```

prints

```
1119
<class 'int'>
```

@yimoj can you make sure that you have merged latest `branch-23.07` into your PR branch?

Reply:
@manopapad I did, which is why the other tests passed, but this one just isn't passing, neither on my local machine nor in the CI runs.

Reply:
OK, it appears the issue is the use of `np.isscalar`, which returns false for 0d arrays.

NumPy will occasionally convert a 0d array to a python-level scalar, but cuNumeric avoids doing that, because in our system an array value can potentially represent an asynchronous computation. Passing that value in its unevaluated form to another operation normally doesn't force the program to block, which we prefer. Converting to a python-level scalar forces blocking, so we don't do it eagerly.

So the general policy here is that we are not going to guarantee that operations which in NumPy return scalars always return scalars in cuNumeric. But the values that we do return should always be usable in place of a regular integer. For example, if the user tries to do anything with the result of `cn.random.randint(1234, size=None)` that requires the value right away, e.g. printing it out, then we block and wait for the computation to complete.

You can instead check that `np.ndim(.) == 0`, which should always be true when a scalar-equivalent must be returned (e.g. in this test, and `test_random_sample_size_none`).